src/overlays/actors/ovl_En_Bsb/z_en_bsb.c

```diff
@@ -124,123 +204,1681 @@ ActorInit En_Bsb_InitVars = {
+        this->drawDmgEffTimer = 0;
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+    }
+
+    func_80C0B31C(play, this, &this->unk_0304);
+    func_80C0B31C(play, this, &this->unk_02F8);
+
+    if (curFrame >= this->animEndFrame) {
+        if ((this->actor.world.pos.z > -1300.0f) || (this->actor.colChkInfo.health < 10)) {
+            func_80C0CA28(this, play);
+        } else {
+            func_80C0C86C(this);
+        }
+    }
+}
+
+void func_80C0D214(EnBsb* this) {
+    this->unk_0294 = 40;
+    this->actor.speed = 0.0f;
+    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_BLUE, 120, COLORFILTER_BUFFLAG_OPA, 40);
+    Actor_PlaySfx(&this->actor, NA_SE_EN_COMMON_FREEZE);
+    this->unk_02B4 = 10;
+    this->actionFunc = func_80C0D27C;
+}
+
+void func_80C0D27C(EnBsb* this, PlayState* play) {
+    if ((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) && (this->drawDmgEffTimer != 0) &&
+        (this->drawDmgEffTimer < 60)) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+    }
+
+    if (((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
+         (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) &&
+        (this->drawDmgEffTimer != 0)) {
+        Actor_SpawnIceEffects(play, &this->actor, this->bodyPartsPos, ENBSB_BODYPART_MAX, 2,
+                              this->drawDmgEffFrozenSteamScale, 0.4f);
+        this->drawDmgEffTimer = 0;
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+    }
+
+    if (this->unk_0294 == 0) {
+        func_80C0C86C(this);
+    }
+}
+
+void func_80C0D334(EnBsb* this) {
+    this->actor.speed = 0.0f;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_KITA_LAUGH_K);
+    EnBsb_ChangeAnim(this, ENBSB_ANIM_6);
+    this->unk_02B4 = 11;
+    this->actionFunc = func_80C0D384;
+}
+
+void func_80C0D384(EnBsb* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    if (curFrame >= this->animEndFrame) {
+        func_80C0C86C(this);
+    }
+}
+
+void func_80C0D3C0(EnBsb* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    Audio_RestorePrevBgm();
+
+    this->unk_02A4 = 0;
+    this->unk_02A8 = 0;
+    this->unk_02AC = 0x4000;
+    this->actor.speed = 0.0f;
+
+    player->actor.world.rot.y = player->actor.shape.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+
+    this->subCamEye.x = this->subCamEyeNext.x = Math_SinS(this->actor.yawTowardsPlayer) * 300.0f + this->unk_02E0.x;
+    this->subCamEye.y = this->subCamEyeNext.y = this->unk_02E0.y - 30.0f;
+    this->subCamEye.z = this->subCamEyeNext.z = Math_CosS(this->actor.yawTowardsPlayer) * 300.0f + this->unk_02E0.z;
+
+    this->subCamAt.x = this->subCamAtNext.x = Math_SinS(this->actor.yawTowardsPlayer) * 10.0f + this->unk_02E0.x;
+    this->subCamAt.y = this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
+    this->subCamAt.z = this->subCamAtNext.z = Math_CosS(this->actor.yawTowardsPlayer) * 10.0f + this->unk_02E0.z;
```

Comment:
```suggestion
    this->subCamEye.x = this->subCamEyeNext.x = this->unk_02E0.x + Math_SinS(this->actor.yawTowardsPlayer) * 300.0f;
    this->subCamEye.y = this->subCamEyeNext.y = this->unk_02E0.y - 30.0f;
    this->subCamEye.z = this->subCamEyeNext.z = this->unk_02E0.z + Math_CosS(this->actor.yawTowardsPlayer) * 300.0f;

    this->subCamAt.x = this->subCamAtNext.x = this->unk_02E0.x + Math_SinS(this->actor.yawTowardsPlayer) * 10.0f;
    this->subCamAt.y = this->subCamAtNext.y = this->unk_02E0.y - 10.0f;
    this->subCamAt.z = this->subCamAtNext.z = this->unk_02E0.z + Math_CosS(this->actor.yawTowardsPlayer) * 10.0f;
```
I think this reads easier